### PR TITLE
Fixed partial rebase issue with forumdisplay

### PR DESCRIPTION
**Pull request by Shade-**
For #2972. This PR fixes a lot of issues due to a partial conversion likely originated from a corrupted/wrong rebase and conflicts fix.

Note that some things are marked as TODO in Twig comments (mainly `get_thread_link` lacking from Twig functions).

### CODE REVIEW

File: inc/views/base/forumdisplay/thread.twig [hunk 1]
```diff
@@ -1,9 +1,10 @@
-{% set class %}{{ thread.bgcolor }} {{ thread.classType }}{% endset %}
+{% set class %}{{ thread.bgcolor }} forumdisplay_{% if thread.sticky %}sticky{% else %}regular{% endif %}{% endset %}
+{% set moved = (thread.moved[0] == 'moved') %}
 <tr class="inline_row">
     <td align="center" class="{{ class }}" width="2%"><span class="thread_status {{ thread.folder.value }}" title="{{ thread.folder.label }}">&nbsp;</span></td>
     <td align="center" class="{{ class }}" width="2%">
         {% if thread.icon and iconCache[thread.icon] %}
-            <img src="{{ iconCache[thread.icon].path }}" alt="{{ iconCache[thread.icon].name }}" title="{{ iconCache[thread.icon].name }}" />
+            <img src="{{ iconCache[thread.icon].path|replace('{theme}', theme.imgdir) }}" alt="{{ iconCache[thread.icon].name }}" title="{{ iconCache[thread.icon].name }}" />
```
Comment (euantorano) [inline inc/views/base/forumdisplay/thread.twig]: Doing this here seems a little bit odd to me, would make sense to do it in the backing code...

File: inc/views/base/forumdisplay/thread.twig [hunk 2]
```diff
@@ -16,36 +17,49 @@
...
+    <td align="center" class="{{ class }}">
+        {% if moved %}
+            -
+        {% else %}
+            <a href="{{ mybb.settings.bburl }}/misc.php?action=whoposted&tid={{ thread.tid }}" onclick="MyBB.whoPosted({{ thread.tid }}); return false;">{{ thread.replies }}</a>
+        {% endif %}
         {% if thread.unapprovedposts and modpermissions.canviewunapprove %}
             <span title="{{ (thread.unapprovedposts == 1) ? trans(lang.thread_unapproved_post_count, 1) : trans(lang.thread_unapproved_posts_count, thread.unapprovedposts) }}">
-                ({{ thread.unapprovedposts }})
+                ({{ thread.unapprovedposts }}){# my_number_format hardcoded, might be worth creating a Twig function #}
```
Comment (euantorano) [inline inc/views/base/forumdisplay/thread.twig]: Note that `my_number_format` is now implemented in #3274.

File: forumdisplay.php [hunk 3]
```diff
@@ -715,466 +710,192 @@
...
+        } elseif ($fpermissions['canviewdeletionnotice'] != 0) {
+            $thread['posts'] += $thread['deletedposts'];
+        }
+
+        if ($thread['posts'] > $mybb->settings['postsperpage']) {
+            $thread['pages'] = $thread['posts'] / $mybb->settings['postsperpage'];
+            $thread['pages'] = ceil($thread['pages']);
+        }
+
+        if($modpermissions['ismod'] && isset($mybb->cookies[$inlinecookie]) && my_strpos($mybb->cookies[$inlinecookie], "|{$thread['tid']}|") !== false)
```
Comment (euantorano) [inline forumdisplay.php]: Code style here should match the other `if` statements.

File: forumdisplay.php [hunk 4]
```diff
@@ -573,13 +570,12 @@
         $cookie = my_unserialize(stripslashes($mybb->cookies['mybb']['announcements']));
     }
 
-    $bgcolor = alt_trow(true); // Reset the trow colors
     while ($announcement = $db->fetch_array($query)) {
         if ($announcement['startdate'] > $mybb->user['lastvisit'] && !$cookie[$announcement['aid']]) {
-            $new_class = ' class="subject_new"';
+            $thread['newclass'] = ' class="subject_new"';
```
Comment (euantorano) [inline forumdisplay.php]: This won't work due to the code in the template:

```twig
<span class="{{ thread.inlineEditClass }} {{ thread.newclass }}" id="tid_{{ thread.tid }}">
```

**Comment by Shade-** [comment]
Fixed ;)

**Pull request merged**